pytensor/tensor/extra_ops.py

```diff
@@ -1561,141 +1561,6 @@ def broadcast_shape_iter(
     return tuple(result_dims)
 
 
-class BroadcastTo(COp):
```

Comment:
`BroadcastTo` is imported in pymc a couple of times. Maybe we should leave an empty Op here, that is deprecated and doesn't do anything?

Reply:
Some of the removed rewrites are also directly imported. 

This shouldn't be a problem however. I marked this PR as a major release so we will bump the version above the upper-bound pinned by PyMC. When we update the pin on PyMC I'll address the changes. They require some manual review though to see if the logic that depended on BroadcastTo was valid per our new rules and can be transferred to Alloc.

This was all on the logprob inference module AFAICT so impact should be pretty contained.

Reply:
sounds good